File: legacy-csharp-avalonia/src/Tools/EyedropperTool.cs
namespace PixelSpriteEditor.Tools;

/// <summary>Picks the color under the cursor and makes it the current color.</summary>
public sealed class EyedropperTool : ITool
{
    public void OnPointerDown(IEditorContext ctx, int x, int y)
    {
        if (!ctx.Canvas.InBounds(x, y)) return;
        ctx.SetCurrentColor(ctx.Canvas.GetPixel(x, y));
    }

    public void OnPointerMove(IEditorContext ctx, int x, int y) => OnPointerDown(ctx, x, y);
    public void OnPointerUp(IEditorContext ctx, int x, int y) { }
}

File: PixelSpriteEditor/src/EyedropperTool.java
/** Picks the color under the cursor and makes it the current color. */
public class EyedropperTool implements Tool {

    @Override
    public void onPress(EditorState state, int x, int y) {
        if (!state.getCanvas().inBounds(x, y)) return;
        state.setCurrentColorArgb(state.getCanvas().getPixel(x, y));
    }

    @Override
    public void onDrag(EditorState state, int x, int y) {
        onPress(state, x, y);
    }

    @Override
    public void onRelease(EditorState state, int x, int y) {
    }
}
